btc/utils.go

```diff
@@ -330,3 +308,34 @@ func getMultiSigAddress(redeemScript string, network string) (string, error) {
 
 	return address.String(), nil
 }
+
```

Comment:
I mean this does not solve the original problem and lower the readability.

It's ok to branch in the previous function and wrap only the isSegwit case in a isolated helper to reuse somewhere else. The isSegwit=false case it's already done just one line `rawAddress, err := key.Address(net)`

Reply:
gotcha